libs/hdf-converters/src/sbom-mapper.ts

```diff
@@ -0,0 +1,482 @@
+import {ExecJSON} from 'inspecjs';
+import _ from 'lodash';
+import {version as HeimdallToolsVersion} from '../package.json';
+import {BaseConverter, ILookupPath, MappedTransform} from './base-converter';
+import {CweNistMapping} from './mappings/CweNistMapping';
+import {getCCIsForNISTTags} from './utils/global';
+
+const CWE_NIST_MAPPING = new CweNistMapping();
+const DEFAULT_NIST_TAG = ['SI-2', 'RA-5'];
+const IMPACT_MAPPING: Map<string, number> = new Map([
+  ['critical', 1.0],
+  ['high', 0.7],
+  ['medium', 0.5],
+  ['low', 0.3],
+  ['info', 0.0],
+  ['none', 0.0],
+  ['unknown', 0.0]
+]);
+
+function formatCWETags(input: number[], addPrefix = true): string[] {
+  const stringifiedCWE: string[] = [];
+  for (const cwe of input) {
+    const cweTag = addPrefix ? `CWE-${cwe}` : `${cwe}`;
+    stringifiedCWE.push(cweTag);
+  }
+  return stringifiedCWE;
+}
+
+function getNISTTags(input: number[]): string[] {
+  return CWE_NIST_MAPPING.nistFilter(
+    formatCWETags(input, false),
+    DEFAULT_NIST_TAG
+  );
+}
+
+// A single SBOM vulnerability can contain multiple security ratings
+// Average any existing ratings and then pass to `impact`
+function aggregateImpact(ratings: Record<string, unknown>[]): number {
+  let impact = 0;
+  for (const rating of ratings) {
+    // Prefer to use CVSS-based `score` field when possible
+    if (_.has(rating, 'score') && _.get(rating, 'method') == 'CVSSv31') {
+      impact += (rating as {score: number}).score;
+    } else {
+      // Else interpret it from `severity` field
+      const severity = IMPACT_MAPPING.get(
+        (rating as {severity: string}).severity.toLowerCase()
+      );
+      if (severity) {
+        impact += severity * 10;
+      }
+    }
+  }
+  // Round up aggregate impact to the 2nd decimal place
+  return Math.ceil((impact / ratings.length) * 10) / 100;
+}
+
+export class SBOMResults {
+  data: Record<string, unknown>;
+  withRaw: boolean;
+  constructor(SBOMJson: string, withRaw = false) {
+    this.data = JSON.parse(SBOMJson);
+    this.withRaw = withRaw;
+
+    if (_.has(this.data, 'components')) {
+      // In-place manipulations on ingested SBOM data
```

Comment:
doing in-place manipulations means that the with-raw isn't actually the raw data.  you can try following the example done in the secure score mapper where an additional variable is created to hold the actual raw data.

Reply:
Addressed.